torchrl/modules/tensordict_module/exploration.py

```diff
@@ -127,10 +127,7 @@ def forward(self, tensordict: TensorDictBase) -> TensorDictBase:
         if exploration_type() == ExplorationType.RANDOM or exploration_type() is None:
             out = tensordict.get(self.td_module.out_keys[0])
             eps = self.eps.item()
-            cond = (torch.rand(tensordict.shape, device=tensordict.device) < eps).to(
-                out.dtype
-            )
-            cond = expand_as_right(cond, out)
+            cond = (torch.rand(out.shape, device=tensordict.device) < eps).to(out.dtype)
```

Comment:
not sure I understand what this does but I don't think it will work with one-hot encoding no?

Reply:
i see, you are right.

The problem here was that if the condition had the shape of the tensordict the agents are left out.

We could do something like
```
cond = (torch.rand(out.shape[:-1], device=tensordict.device) < eps).to(out.dtype)
cond = expand_as_right(cond, out)
```
which would work with:
`MultiDiscrete, OneHot, MultiOneHot` 
since they have the hardcoded that the last dimension is the action dimension.

It would not work with `DiscreteTensorSpec`since here the last dimension would be the agents.

Should we uniform `DiscreteTensorSpec` to the other discrete specs ( by making it have the last dim of 1) and go for this solution?

Reply:
unifroming `DiscreteTensorSpec ` would also get rid of some unsqueezes like here https://github.com/pytorch/rl/blob/main/torchrl/modules/tensordict_module/actors.py#L490